themes/default/subtemplates/entry.inc.tpl

```diff
@@ -21,7 +21,9 @@
 {else}
 {assign var=name value="$name"}
 {/if}
-{if $posting_user_id>0 && ($user || $settings.user_area_public==1)}{assign var=name value="<a href=\"index.php?mode=user&amp;show_user=$posting_user_id\">$name</a>"}{/if}
+{if (($settings.user_area_access == 0 and ($admin or $mod)) or ($settings.user_area_access == 1 and $user) or $settings.user_area_access == 2) && $posting_user_id>0}
```

Comment:
Neither in the template (`entry.inc.tpl`) itself nor in the underlying `entry.inc.php`, `index.inc.php` or the main script `index.php` I see a definition of `$user`, `$mod` or `$admin`. **But …**

I tested the code in my testing forum with my admin account, with a new testing account as normal user and later as a moderator. In every (user wise) case with every possible setting I encountered the expected behaviour. In this case it would be `user_area_access = 1` what should allow registered users and also admins and mods to access the user list page. It works for registerd users, admins and mods and prevents the access for unregistered/not logged in visitors. Same tests for `user_area_access = 0` and `user_area_access = 2` succeeded. So it works as expected.